### PR TITLE
Updated and fixed all syntax errors and warnings

### PR DESCRIPTION
**Pull request by DazKins**
A lot of the terraform code here is old and it will spit out a bunch of errors/warnings when you try to run it

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,9 +1,9 @@
 terraform {
-  required_version = "~>0.11"
+  required_version = "0.12.18"
```
Comment (vincentsarago) [inline main.tf:2]: @DazKins To be honest I've haven't used TF for a long time, setting required version down to minor version might be a bit strong? could this be something like `~0.12` ?

**Comment by vincentsarago** [comment]
hey @DazKins this PR is really appreciated. As mentioned I haven't used TF for a really long time (since update broke all my dependencies 😬). I'm ok merging this but I have no way to tests if it will work or not. 
Do you know if there are ways to tests TF code ?

File: main.tf (hunk 1 above)
Comment (DazKins) [reply main.tf]: Sounds good. Changed now

**Comment by DazKins** [comment]
> hey @DazKins this PR is really appreciated. As mentioned I haven't used TF for a really long time (since update broke all my dependencies 😬). I'm ok merging this but I have no way to tests if it will work or not.
> Do you know if there are ways to tests TF code ?

Hey, I'm not to sure about testing terraform code. I've seen a few articles online about some people doing it. We are not currently testing our terraform deployment code.

I have been playing around with module using it for our lambda deployments however and it all seems to be working fine

**Comment by vincentsarago** [comment]
Thanks a lot for this PR @DazKins 🙏

**Pull request merged**